content/en/docs/components/pipelines/sdk/connect-api.md

```diff
@@ -4,227 +4,337 @@ description = "How to connect to Kubeflow Pipelines using the SDK client and con
 
-Kubeflow Pipelines standalone deploys a Kubernetes service named `ml-pipeline-ui` in your Kubernetes cluster without extra authentication.
+Kubeflow Pipelines (standalone) deploys an unsecured Kubernetes `Service` named `ml-pipeline-ui` which requires no authentication.
 
-You can use [kubectl port-forward](https://kubernetes.io/docs/tasks/access-application-cluster/port-forward-access-application-cluster/) to port forward the Kubernetes service locally to your laptop outside of the cluster:
+Use [kubectl port-forward](https://kubernetes.io/docs/tasks/access-application-cluster/port-forward-access-application-cluster/) to forward the service locally to your laptop:
 
 ```bash
-# Change the namespace if you deployed Kubeflow Pipelines in a different
-# namespace.
-$ kubectl port-forward svc/ml-pipeline-ui 3000:80 --namespace kubeflow
+# change the namespace if you deployed Kubeflow Pipelines in a different namespace.
+kubectl port-forward --namespace kubeflow svc/ml-pipeline-ui 3000:80
 ```
 
-You can verify that port forwarding is working properly by visiting [http://localhost:3000](http://localhost:3000) in your browser. If port forwarding is working properly, the Kubeflow Pipelines UI appears.
+You can verify that port forwarding is working properly by visiting [http://localhost:3000](http://localhost:3000) in your browser. 
+If working properly, the Kubeflow Pipelines UI will appear.
 
-Run the following python code to instantiate the `kfp.Client`:
+Run the following python code to instantiate a `kfp.Client()` against your port-forwarded Service:
 
 ```python
 import kfp
+
 client = kfp.Client(host='http://localhost:3000')
 print(client.list_experiments())
 ```
 
-Note, for Kubeflow Pipelines in multi-user mode, you cannot access the API using kubectl port-forward
-because it requires authentication. Refer to distribution specific documentation as recommended above.
+### Multi-User mode
 
-## Connect to Kubeflow Pipelines from the same cluster
+For Kubeflow Pipelines in multi-user mode, the process to access the Pipelines API will vary depending on your distribution:
+* [Google Cloud](/docs/distributions/gke/pipelines/authentication-sdk/)
+* [AWS](/docs/distributions/aws/pipeline/#authenticate-kubeflow-pipeline-using-sdk-inside-cluster)
+* [Azure](/docs/distributions/azure/authentication-oidc/)
+* [IBM Cloud Kubernetes Service (IKS)](/docs/distributions/ibm/pipelines/)
 
-### Non-multi-user mode
+__Example for deployments that use [dex](https://dexidp.io/).__
+
+First you expose your `Service/istio-ingressgateway` locally:
+```bash
+# NOTE: `svc/istio-ingressgateway` may be called something else, or use different ports
+kubectl port-forward --namespace istio-system svc/istio-ingressgateway 8080:80
+```
+
+This Python function will return a session cookie by authenticating with dex:
+
+```python
+import re
+import requests
+from urllib.parse import urlsplit
+
+def get_istio_auth_session(url: str, username: str, password: str) -> dict:
```

Comment:
I don't really know of a better place to put this.

Also, since dex is used in the default `kubeflow/manifests`, it probably doesn't hurt to include it here.

Reply:
This was also discussed in https://github.com/kubeflow/website/issues/2916 where @Bobgy agreed (https://github.com/kubeflow/website/issues/2916#issuecomment-914265013) it makes sense to include in the base pipelines docs.

Because Dex is used by so many Kubeflow distributions.

Reply:
Sounds good on having the code snippet in the doc then. Hope that it doesn't create too much difficulty in keeping the code snippet up-to-date.